zmq/protorequester/protorequester.cpp

```diff
@@ -173,7 +179,7 @@ void zmq::ProtoRequester::ProcessRequest(zmq::socket_t& socket, RequestStruct& r
             std::runtime_error error("ZMQ ProtoRequester was shutdown during request handling");
             request.promise.set_exception(std::make_exception_ptr(error));
         }
-    }catch(const std::exception& ex){
+    }catch(const std::exception&){
```

Comment:
Go back to using the exception reference.

Reply:
Turns out that the promise wants an exception_ptr, so current_exception is the more convenient method for doing this.